pallets/moonbeam-foreign-assets/src/lib.rs

```diff
@@ -225,6 +253,24 @@ pub mod pallet {
 	pub type AssetsByLocation<T: Config> =
 		StorageMap<_, Blake2_128Concat, Location, (AssetId, AssetStatus)>;
 
+	/// Mapping from an asset id to its creation details
+	#[pallet::storage]
+	#[pallet::getter(fn assets_creation_details)]
+	pub type AssetsCreationDetails<T: Config> =
+		StorageMap<_, Blake2_128Concat, AssetId, AssetCreationDetails<T>>;
+
+	#[derive(Clone, Eq, PartialEq, Debug, Encode, Decode, TypeInfo, MaxEncodedLen)]
+	pub enum AssetOwner<T: Config + scale_info::TypeInfo> {
```

Comment:
`AccountId` is bounded by `Parameter` trait, which already enforces `TypeInfo`.

> pub trait Parameter: Codec + EncodeLike + Clone + Eq + Debug + TypeInfo {}

```suggestion
	pub enum AssetOwner<T: Config> {
```